Description: Prevent unauthorized sudoedits
 Fix an issue where users with limited sudoedit rights could exploit a race
 condition to get escalated sudoedit permissions.
Origin: upstream, embargoed
Forwarded: not-needed
Author: Marc Schoolderman <<REDACTED_EMAIL>>
Bug: https://bugs.launchpad.net/ubuntu/+source/rust-sudo-rs/+bug/2165142
Last-Update: 2026-08-25

diff --git a/src/common/context.rs b/src/common/context.rs
index e0621f7f..9572a798 100644
--- a/src/common/context.rs
+++ b/src/common/context.rs
@@ -126,17 +126,19 @@ impl Context {
                 .map_err(|_| arg)
                 .and_then(|path| path.into_os_string().into_string().map_err(|_| arg))
             })
+            .collect::<Vec<_>>()
         })?;
 
         let files_to_edit = resolved_args
-            .clone()
-            .map(|path| path.ok().map(SudoPath::from_cli_string))
+            .iter()
+            .map(|path| path.clone().ok().map(SudoPath::from_cli_string))
             .collect();
 
         // if a path resolved to something that isn't in UTF-8, it means it isn't in the sudoers file
         // as well and so we treat it "as is" wrt. the policy lookup and fail if the user is allowed
         // by the policy to edit that file. this is to prevent leaking information.
         let arguments = resolved_args
+            .into_iter()
             .map(|arg| match arg {
                 Ok(arg) => OsString::from(arg),
                 Err(arg) => OsString::from(arg),
